src/main/java/org/springframework/integration/kafka/listener/KafkaTopicOffsetManager.java

```diff
@@ -0,0 +1,505 @@
+ * limitations under the License.
+ */
+
+
+package org.springframework.integration.kafka.listener;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import kafka.admin.AdminUtils$;
+import kafka.api.OffsetRequest;
+import kafka.common.ErrorMapping$;
+import kafka.common.TopicExistsException;
+import kafka.javaapi.producer.Producer;
+import kafka.producer.DefaultPartitioner;
+import kafka.producer.KeyedMessage;
+import kafka.serializer.Decoder;
+import kafka.serializer.Encoder;
+import kafka.utils.ZKStringSerializer$;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.exception.ZkInterruptedException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.BeanInitializationException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.integration.kafka.core.BrokerAddress;
+import org.springframework.integration.kafka.core.ConnectionFactory;
+import org.springframework.integration.kafka.core.FetchRequest;
+import org.springframework.integration.kafka.core.KafkaMessage;
+import org.springframework.integration.kafka.core.KafkaMessageBatch;
+import org.springframework.integration.kafka.core.KafkaTemplate;
+import org.springframework.integration.kafka.core.Partition;
+import org.springframework.integration.kafka.core.Result;
+import org.springframework.integration.kafka.core.TopicNotFoundException;
+import org.springframework.integration.kafka.support.ProducerFactoryBean;
+import org.springframework.integration.kafka.support.ProducerMetadata;
+import org.springframework.integration.kafka.support.ZookeeperConnect;
+import org.springframework.integration.kafka.util.LoggingUtils;
+import org.springframework.integration.kafka.util.MessageUtils;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.util.Assert;
+import org.springframework.util.ObjectUtils;
+
+/**
+ * Implementation of an {@link OffsetManager} that uses a Kafka topic as the underlying support.
+ * For its proper functioning, the Kafka server(s) must set {@code log.cleaner.enable=true}. It relies on the property
+ * {@code cleanup.policy=compact} to be set on the target topic, and if the topic is not found,
+ * it will create a topic with the appropriate settings.
+ *
+ * @author Marius Bogoevici
+ */
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class KafkaTopicOffsetManager extends AbstractOffsetManager implements InitializingBean {
```

Comment:
In practice? it is typical SI application behaviour, when we manage our `Lifecycles`. ControlBus is standard component to do that:

````xml
<control-bus channel="controlBusChannel"/>
````
````java
this.controlBusChannel.send(new GenericMessage<String>("@myKafkaAdapter.stop()"));
````